CodeModel/src/main/java/org/openzen/zenscript/codemodel/expression/Expression.java

```diff
@@ -142,8 +141,11 @@ public Expression call(CodePosition position, TypeScope scope, List<TypeID> hint
 
 	@Override
 	public IPartialExpression getMember(CodePosition position, TypeScope scope, List<TypeID> hints, GenericName name) throws CompileException {
-		TypeMembers members = scope.getTypeMembers(type);
-		IPartialExpression result = members.getMemberExpression(position, scope, this, name, false);
+		IPartialExpression result = scope.getTypeMembers(type).getMemberExpression(position, scope, this, name, false);
+		// If the type is optional, check if the member exists on the normal type.
+		if (result == null && type.isOptional())
+			result = scope.getTypeMembers(type.withoutOptional()).getMemberExpression(position, scope, this, name, false);
+
```

Comment:
The logic makes sense, but I believe it is not at the right place. I believe this should be in the TypeMembers class instead.

There might be other code that retrieves a member, and putting the logic here would cause type member resolution to be inconsistent.